include/ghex/transport_layer/ucx/context.hpp

```diff
@@ -171,73 +178,78 @@ namespace gridtools {
                     if (!m_workers[t.id()])
                     {
                         m_tokens[t.id()] = t;
-                        m_workers[t.id()] = std::make_unique<worker_type>(this, &m_parallel_context, &m_tokens[t.id()], UCS_THREAD_MODE_SINGLE);
+                        m_workers[t.id()] = std::make_unique<worker_type>(this, &m_thread_primitives, &m_tokens[t.id()], UCS_THREAD_MODE_SINGLE);
                     }
                     return {&m_worker, m_workers[t.id()].get()};
                 }
-                    
+
                 rank_type rank() const { return m_db.rank(); }
                 rank_type size() const { return m_db.size(); }
                 ucp_context_h get() const noexcept { return m_context.m_context; }
 
+                void barrier()
+                {
+                }
```

Comment:
can you come up with an implementation of the barrier? do we need it for the benchmarks/tests?

Reply:
done